application/ipc/electron.ipc.messages/file.filepicker.response.ts

```diff
@@ -0,0 +1,35 @@
+
+export interface IFilePickerFileInfo {
+    path: string;       // Full file name (path + basename)
+    name: string;       // Only basename
+    size: number;       // Size in bytes
+    created: number;    // unixtime
+    changed: number;    // unixtime
+}
+
+export interface IFilePickerResponse {
+    files: IFilePickerFileInfo[];
+    error?: string;
+}
+
+export class FilePickerResponse {
+
+    public static signature: string = 'FilePickerResponse';
+    public signature: string = FilePickerResponse.signature;
+    public files: IFilePickerFileInfo[] = [];
+    public error: string | undefined;
+
+    constructor(params: IFilePickerResponse) {
+        if (typeof params !== 'object' || params === null) {
+            throw new Error(`Incorrect parameters for FilePickerResponse message`);
+        }
+        if (!(params.files instanceof Array)) {
+            throw new Error(`files should be defined as IFilePickerFileInfo[].`);
+        }
+        if (params.error !== undefined && typeof params.error !== 'string') {
+            throw new Error(`error can be defined only as string`);
+        }
```

Comment:
those type assertions should not be needed. we know what the type of params is. so why add those checks?

Reply:
Because we have to check also it in run-time. Simple use-case, some information comes from `indexer-neon`, we are expecting field `params.files` will be an array. And during compilation - no any issues. But in run-time, something goes wrong and instead of an array we've gotten `undefined`. 

Without run-time checks, such error will go forward and fires some bug, for example, on render level. So, developer will start to debug somewhere in front-end, while problem comes from another place.

BTW, a time ago, we had exactly same use-case: in run-time indexer-neon sends wrong data and because we have such checks in scope of IPC message-classes, issue was located very fast